src/components/Modal/Modal.md

```diff
@@ -0,0 +1,79 @@
+## Description
+The `<Modal />` component provides a container that renders content into an accessible modal outlet. By default, the container will render `children` centered in the screen with an overlay. Out of the box, the `<Modal />` will handle setting proper focus on tabable elements within the inner content, trapping focus within the `modal`,  and toggle the state of the modal through keyboard events.
+
+The container makes no assumptions about the inner content and can be styled to fit whatever implementation is needed.
+
+## Examples
+
+### Basic Usage
+Activating the modal is done through passing a `boolean` flag as the `showModal` `prop`. This gives the parent the ability to set if the modal will render open or closed on initial render.
+
+Toggling the state of the modal is also handled by passing a callback through the `toggle` `prop`. The callback is a `voided` function that accepts a  `boolean` as whether the modal is visible. When either the `escape` key is hit, or outside of the modal content is clicked, the modal will pass a boolean value of `false` to the `toggle` method.
```

Comment:
The part:

> is also handled by

confused me; I think `toggle` may be a slightly confusing name since there are two things happening (I think):

- Toggle the modal open or closed (action)
- Toggle the state (question e.g. is it open or closed?)

My understanding of the code is that:

- `showModal` will perform the action of showing/hiding the modal based on the boolean arg any time the Modal is rendered
- `toggle` is a callback that specifically receives a status boolean indicating whether the modal is visible or not

Is there a reason to pass in a callback other than a state hook setter? Should we name this more specifically to reduce confusion e.g. `toggleIsOpen` or something?

Reply:
I agree, as Tamir pointed out above, internally, the modal only ever passes `false` to the `toggle` method. It would make sense to name this `prop` something like `onDismiss`. `showModal` could be renamed to `isActive` or `isShown`

There might be times where you might want to pass a function to `toggle` that's not just a state setting, e.g. maybe when you dismiss the modal, you want to fire some sort of tracking event.

Reply:
Yeah, on second read `onDismiss` sounds like a winner 👍